### PR TITLE
Feature/weather forecast

### PR DESCRIPTION
**Pull request by dsj976**
New bit of code to upload hourly weather forecast data (up to 48h into the future) to a new DB table called "weather_forecast".
I have tested the code locally and it is uploading data to the DB without any issues.
I am not sure if I have done the Azure function/Docker/GitHub Action procedure correctly, so please have a look at this.

### CODE REVIEW

**Review by nbarlowATI — COMMENTED** [review]
Hi David,
(as just discussed in person) - it could be good to store both the time of the request (as `timestamp`) and the time for which the forecast is being made (as something like `forecast_time`).     
For the timestamp, in `structure.py` there are a couple of tables where a column is defined as something like 
```
Column(DateTime(), server_default=func.now())
```
which might do what we want?  (let sqlalchemy/postgres do the hard work :) )

File: core/sensors.py [hunk 1]
```diff
@@ -147,6 +148,33 @@ def get_db_weather_data(session, date_from, date_to):
...
+
+    Arguments:
+        session: sqlalchemy active session object
+        date_from: date range from
+        date_to: date range to
+    Returns:
+        result_df: data frame containing sensor data for specific period of time
+    """
+
+    query = session.query(WeatherForecastsClass.timestamp,).filter(
```
Comment (mhauru) [inline core/sensors.py:168]: The trailing comma reminded me: We don't have a strict policy that code must be formatted using black (if you're not familiar: https://github.com/psf/black), but most of it is, and we'll occasionally run black over all the files. I format everything I write with black, Nick I think doesn't, although I'm not sure about that.

**Review by mhauru — COMMENTED** [review]
I don't really have anything to add to what Nick said: Good to have both timestamps, otherwise looks good.

File: core/sensors.py (hunk 1 above)
Comment (dsj976) [reply core/sensors.py:168]: So apparently the *weather_forecast* table **is logging** the time of the request (the column is called *time_created*). This is not actually the time of the request to the openweathermap API, but rather the time at which the row is created in the table - they should be very similar in any case (I actually did include `time_created = Column(DateTime(), server_default=func.now())` in the SQLAlchemy class).

What I don't understand is why the *iweather* table has Null entries for *time_created*, when it also has `Column(DateTime(), server_default=func.now())` in its SQLAlchemy class....

Not sure how you would want to deal with this, but given that the forecast time is labelled *timestamp* in *iweather* (and I am assuming the GES model is coded up to read this column from the table), I guess it makes sense to stick with this same convention in *weather_forecast*?

**Comment by dsj976** [comment]
So apparently the *weather_forecast* table **is logging the time of the request** (the column is called *time_created*). This is not actually the time of the request to the openweathermap API, but rather the time at which the row is created in the table - they should be very similar in any case (I actually did include `time_created = Column(DateTime(), server_default=func.now())` in the SQLAlchemy class).

What I don't understand is why the *iweather* table has Null entries for *time_created*, when it also has `Column(DateTime(), server_default=func.now())` in its SQLAlchemy class....

Not sure how you would want to deal with this, but given that the forecast time is labelled *timestamp* in *iweather* (and I am assuming the GES model is coded up to read this column from the table), I guess it makes sense to stick with this same convention in *weather_forecast*?

**Comment by nbarlowATI** [comment]
OK great, yep, makes sense, let's keep the column names consistent between the weather forecast and past weather data.   
So I think this is good to merge :)

**Pull request merged**